py/private/py_executable.bzl

```diff
@@ -26,7 +26,7 @@ def _determine_main(ctx):
     """
     if ctx.attr.main:
         # Deviation from rules_python: allow a leading colon, e.g. `main = ":my_target"`
-        proposed_main = ctx.attr.main.removeprefix(":")
+        proposed_main = ctx.attr.main.label.name.removeprefix(":")
```

Comment:
I don't think `label.name` will ever have a  `:` as prefix. maybe we are really missing something here?